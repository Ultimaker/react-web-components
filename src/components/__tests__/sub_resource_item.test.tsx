// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import SubResourceItem, { SubResourceItemProps } from '../sub_resource_item';

describe('The SubResourceItem component', () => {
    let props: SubResourceItemProps;
    let wrapper;

    beforeEach(() => {
        props = {
            subResource: {
                subResourceId: '1',
                name: 'Test',
                showName: true,
            },
        };
        wrapper = shallow(<SubResourceItem {...props} />);
    });

    it('should render names', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render icons', () => {
        wrapper.setProps({
            subResource: {
                subResourceId: '1',
                name: 'Test',
                showIcon: true,
                iconUrl: '../../images/logobot-placeholder-dark.svg',
            },
        });
        expect(wrapper).toMatchSnapshot();
    });
});
