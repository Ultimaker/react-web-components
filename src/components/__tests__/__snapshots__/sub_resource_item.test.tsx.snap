// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SubResourceItem component should render icons 1`] = `
<Tooltip
  direction="north"
  tooltipText="Test"
>
  <div
    className="sub-resource-item"
  >
    <ProfileImage
      imageURL="../../images/logobot-placeholder-dark.svg"
      size="3rem"
    />
  </div>
</Tooltip>
`;

exports[`The SubResourceItem component should render names 1`] = `
<Tooltip
  direction="north"
  tooltipText="Test"
>
  <div
    className="sub-resource-item"
  >
    <Pill
      active={false}
    >
      Test
    </Pill>
  </div>
</Tooltip>
`;
